app/jobs/search/index_job.rb

```diff
@@ -0,0 +1,14 @@
+module Search
+  class IndexJob < ApplicationJob
+    queue_as :search_index
+
+    def perform(record_type, record_id)
```

Comment:
100% agree this is the way to go passing a type and id rather than an ActiveRecord object!